services/src/main/java/org/keycloak/authorization/authorization/AuthorizationTokenService.java

```diff
@@ -223,17 +241,23 @@ public Response authorize(KeycloakAuthorizationRequest request) {
                     } else if (RESPONSE_MODE_PERMISSIONS.equals(metadata.getResponseMode())) {
                         return createSuccessfulResponse(permissions, request);
                     } else {
-                        throw new CorsErrorResponseException(request.getCors(), OAuthErrorException.INVALID_REQUEST, "Invalid response_mode", Status.BAD_REQUEST);
+                        CorsErrorResponseException invalidResponseModeException = new CorsErrorResponseException(request.getCors(), OAuthErrorException.INVALID_REQUEST, "Invalid response_mode", Status.BAD_REQUEST);
+                        fireErrorEvent(event, Errors.INVALID_RESPONSE_MODE, invalidResponseModeException);
+                        throw invalidResponseModeException;
                     }
                 } else {
                     return createSuccessfulResponse(createAuthorizationResponse(identity, permissions, request, targetClient), request);
                 }
             }
 
             if (request.isSubmitRequest()) {
-                throw new CorsErrorResponseException(request.getCors(), OAuthErrorException.ACCESS_DENIED, "request_submitted", Status.FORBIDDEN);
+                CorsErrorResponseException submittedRequestException = new CorsErrorResponseException(request.getCors(), OAuthErrorException.ACCESS_DENIED, "request_submitted", Status.FORBIDDEN);
+                fireErrorEvent(event, Errors.INVALID_REQUEST, submittedRequestException);
```

Comment:
Not really, an invalid request. Here the access is denied by the request was submitted for approval by another user (usually the resource owner).

Reply:
using Errors.ACCESS_DENIED would be more appropriate then?

Reply:
Yes. Because that is what happened. You might argue though that you want to audit attempts to access users' resources. So maybe a `APPROVAL_REQUEST` makes sense?